components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7755,17 +7724,20 @@ private CommentList getComments(Identifier identifier, String parentCommentID, I
         int total = 0;
         String sqlQuery;
         String sqlQueryForCount;
+        int id = getAPIIDFromUUID(uuid);
```

Comment:
May be we can modify the Query to get replies using UUID instead of API ID. Otherwise there will be one additional query. In master branch we used UUID. You can refer this one. https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java#L7664